src/txn/txn.c

```diff
@@ -318,6 +318,58 @@ __wt_txn_bump_snapshot(WT_SESSION_IMPL *session)
     __txn_get_snapshot_int(session, false);
 }
 
+/*
+ * __wt_txn_copy_and_bump_snapshot --
+ *     Take backup of the existing snapshot and allocate a new snapshot.
+ */
+void
+__wt_txn_copy_and_bump_snapshot(WT_SESSION_IMPL *session)
+{
+    WT_TXN *txn;
+
+    txn = session->txn;
+
+    txn->bkp_snap_max = txn->snap_max;
+    txn->bkp_snap_min = txn->snap_min;
+    txn->bkp_snapshot_count = txn->snapshot_count;
+
+    /* Take the backup of the snapshot only if the snapshot count is greater than 0. */
+    if (txn->snapshot_count > 0) {
+
+        WT_IGNORE_RET(
+          __wt_calloc(session, txn->snapshot_count, sizeof(uint64_t), &txn->bkp_snapshot));
+        memcpy(txn->bkp_snapshot, txn->snapshot, txn->snapshot_count * sizeof(uint64_t));
```

Comment:
we can avoid the memcpy, as snapshot is just pointer, exchange the pointers instead.

Reply:
Fixed in 2a2129410c9c3f8ac7428e3afdb57c2eaad6febb